src/internal/connector/data_collections.go

```diff
@@ -61,20 +61,6 @@ func (gc *GraphConnector) ProduceBackupCollections(
 		return nil, nil, clues.Stack(err).WithClues(ctx)
 	}
 
-	serviceEnabled, canMakeDeltaQueries, err := checkServiceEnabled(
```

Comment:
I think it's still sensible to make this validation within the backup producer.  Or, at least, suppose the result of the call from IsBackupRunnable gets cached.  In the (unlikely, but possible) event that ProduceBackupCollections cannot find the cached results, it should perform the lookup.

This is primarily to ensure behavior is correct across interface boundaries.  We _assume_ that IsBackupRunnable is called before ProduceBackupCollections.  But that's not a precondition; the code does not require it to have been called.  As a result we can either re-run the check as a fallback, or we could error out if the isRunnable cache is unpopulated, and make it a precondition.